### PR TITLE
[Core] Reduce node and dof size stage3

### PR DESCRIPTION
**Pull request by pooyan-dadvand**
This PR reduces the Dof size from **32 bytes** to **16 bytes**. 

The behavior change comes from the restriction that we cannot add a dof to a node in a parallel section. To do that we should have been added that dof to the ModelPart. (No direct interface yet) The VariablesUtils::AddDof has changed accordingly so it can be used safely. Also, a debug error will alert in case of non-thread-safe usage.

A new restriction is also imposed by this PR: It is **not possible to store more than 64 Dofs per node**. A debug error is also added to reflect this restriction.

### CODE REVIEW

File: kratos/containers/variables_list.h
Comment (loumalouomega) [inline kratos/containers/variables_list.h:311]: Don't you prefer to use KRATOS_ERROR_IF directly?, it is clearer

Comment (pooyan-dadvand) [reply kratos/containers/variables_list.h:311]: It was to avoid the include in release mode as it was before. You are right that the other way is preferred.

Comment (philbucher) [reply kratos/containers/variables_list.h:311]: I think he meant `KRATOS_ERROR_IF(OpenMPUtils::IsInParallel() != 0)`

Comment (pooyan-dadvand) [reply kratos/containers/variables_list.h:311]: I know, I just didn't want to change how the omp_utils.h is included.

**Comment by philbucher** [comment]
now travis compiles

File: kratos/tests/cpp_tests/sources/test_node.cpp
Comment (philbucher) [inline kratos/tests/cpp_tests/sources/test_node.cpp]:
```suggestion
```

File: kratos/tests/cpp_tests/sources/test_node.cpp
Comment (philbucher) [inline kratos/tests/cpp_tests/sources/test_node.cpp]:
```suggestion
```

File: kratos/includes/dof.h
Comment (philbucher) [inline kratos/includes/dof.h:560]: why is the direct load not possible here?

Comment (pooyan-dadvand) [reply kratos/includes/dof.h:560]: Because I wasn't sure if it takes the correct type between all overloaded ones. So I choose it explicitly.

**Comment by rubenzorrilla** [comment]
@Rbravo555 could you please try that this PR doesn't introduce any issues with the Raspberry Pi?

**Review by philbucher — APPROVED** [review]
for my cases/tests it works
also the test in the apps that I compile are working fine
Hence I am approving but I am not sure if you also want to wait for the reviews of others

**Comment by Rbravo555** [comment]
> @Rbravo555 could you please try that this PR doesn't introduce any issues with the Raspberry Pi?

After compiling the branch in the Raspberry Pi, both the Core tests, as well as the ones of the ConvectionDiffusionApplication are succesfully passed.

**Comment by loumalouomega** [comment]
> 
> 
> > @Rbravo555 could you please try that this PR doesn't introduce any issues with the Raspberry Pi?
> 
> After compiling the branch in the Raspberry Pi, both the Core tests, as well as the ones of the ConvectionDiffusionApplication are succesfully passed.

Do we need #5585 then?

**Pull request merged**
